### PR TITLE
Enable gce pd driver via a flag rather than an env var

### PR DESCRIPTION
**Pull request by mattcary**
See https://github.com/kubernetes-sigs/kubetest2/issues/202 for context. 
/kind cleanup

/sig-storage

```release-note
Add e2e test flag to specify which volume drivers should be installed. This deprecates the ENABLE_STORAGE_GCE_PD_DRIVER environment variable.
```

### CODE REVIEW

File: test/e2e/framework/test_context.go [hunk 1]
```diff
@@ -319,6 +322,8 @@ func RegisterCommonFlags(flags *flag.FlagSet) {
 
 	flags.StringVar(&TestContext.SnapshotControllerPodName, "snapshot-controller-pod-name", "", "The pod name to use for identifying the snapshot controller in the kube-system namespace.")
 	flags.IntVar(&TestContext.SnapshotControllerHTTPPort, "snapshot-controller-http-port", 0, "The port to use for snapshot controller HTTP communication.")
+
+	flags.BoolVar(&TestContext.EnableGcePdDriver, "enable-gcepd-driver", false, "Enable the gcepd volume driver. This should be used only on GCE clusters where the PD CSI driver is installed. See https://github.com/kubernetes/cloud-provider-gcp/tree/master/cluster/addons/pdcsi-driver for an example.")
```
Comment (msau42) [inline test/e2e/framework/test_context.go:326]: can you make this a generic arg that takes a comma separated string of driver names? I'm thinking we may want to move all the platform-specific drivers to be off by default in the future.

**Review by msau42 — COMMENTED** [review]
Can you also add a release note with the flag details and also saying that it removes the env var that was added in the other PR?

File: test/e2e/storage/in_tree_volumes.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ var testDrivers = []func() storageframework.TestDriver{
 
 // This executes testSuites for in-tree volumes.
 var _ = utils.SIGDescribe("In-tree Volumes", func() {
-	if enableGcePD := os.Getenv("ENABLE_STORAGE_GCE_PD_DRIVER"); enableGcePD == "yes" {
+	if TestContext.EnableGcePdDriver {
```
Comment (msau42) [inline test/e2e/storage/in_tree_volumes.go:56]: Maybe for now, support both env var and flag to give people time to fix their CI config.

**Comment by msau42** [comment]
fyi @jsafrane @bertinatto @gnufied

File: test/e2e/framework/test_context.go (hunk 1 above)
Comment (mattcary) [reply test/e2e/framework/test_context.go:326]: Good idea.

File: test/e2e/storage/in_tree_volumes.go (hunk 2 above)
Comment (mattcary) [reply test/e2e/storage/in_tree_volumes.go:56]: Sure (although I'd be willing to bet that the only configs changed were the kubetest2 ones I did where this wasn't actually effective...)

**Comment by mattcary** [comment]
I implemented your changes. I'll work on testing this manually to make sure the plumbing actually works with kubetest2 (which is the ultimate goal here).

/hold

File: test/e2e/framework/test_context.go [hunk 3]
```diff
@@ -319,6 +343,8 @@ func RegisterCommonFlags(flags *flag.FlagSet) {
 
 	flags.StringVar(&TestContext.SnapshotControllerPodName, "snapshot-controller-pod-name", "", "The pod name to use for identifying the snapshot controller in the kube-system namespace.")
 	flags.IntVar(&TestContext.SnapshotControllerHTTPPort, "snapshot-controller-http-port", 0, "The port to use for snapshot controller HTTP communication.")
+
+	flags.Var(&stringArrayValue{&TestContext.EnabledVolumeDrivers}, "enabled-volume-drivers", "Comma-separated list of volume drivers which are disabled by default. An exmaple is gcepd; see test/e2e/storage/in_tree_volumes.go for full details.")
```
Comment (msau42) [inline test/e2e/framework/test_context.go:347]: wording suggestion: "Comma-separated list of intree volume drivers to run tests against".

File: test/e2e/storage/in_tree_volumes.go (hunk 2 above)
Comment (msau42) [reply test/e2e/storage/in_tree_volumes.go]: I noticed openshift updated CI to use the envs.

**Comment by msau42** [comment]
nit on release note wording. The env var is deprecated (instead of removed)

**Comment by mattcary** [comment]
> nit on release note wording. The env var is deprecated (instead of removed)

good catch, thx

File: test/e2e/framework/test_context.go (hunk 3 above)
Comment (mattcary) [reply test/e2e/framework/test_context.go:347]: done

**Comment by mattcary** [comment]
/unhold

Tested as best I could, seems to work.  Although I'm not sure where the framework.Log messages go to,

**Pull request merged**